blockchains/erc20/erc20_worker.js

```diff
@@ -63,17 +78,28 @@ class ERC20Worker extends BaseWorker {
 
     logger.info(`Fetching transfer events for interval ${result.fromBlock}:${result.toBlock}`);
 
-    let events;
+    let events = [];
     let overwritten_events = [];
     if ('extract_exact_overwrite' === constants.CONTRACT_MODE) {
-      events = await contractEditor.getPastEventsExactContracts(this.web3, result.fromBlock, result.toBlock,
-        new TimestampsCache());
-      contractEditor.changeContractAddresses(events);
+      const timestampsCache = new TimestampsCache();
+      for (const contractOverwrite of this.contractsOverwriteArray) {
```

Comment:
I am changing the logic of iteration a bit here. Before we called `contractEditor.getPastEventsExactContracts` and getting the events for all the contracts. We did not bulk the calls (see removed code), instead iterated per contract. However the information for which events are for which contract was lost, then we had to do some logic to map events to contract so to overwrite. This is not effective and somewhat harder to comprehend.